stacks-signer/src/v0/signer.rs

```diff
@@ -53,10 +57,14 @@ pub struct Signer {
     pub signer_addresses: Vec<StacksAddress>,
     /// The reward cycle this signer belongs to
     pub reward_cycle: u64,
+    /// Reward set signer addresses and their weights
+    pub signer_weights: HashMap<StacksAddress, usize>,
```

Comment:
With this change, should we just drop `signer_addresses` and maybe use an `IndexMap` to preserve the order we want?

Reply:
So, the reason I'm using `.find` is because we only want to check the address bytes, not the address version.  This removes a category of bugs that could arise from needing to preserve access to the `.mainnet` bool pervasively in the code.  In order for the above code to work, we'd need to construct the `StacksAddress` with `.mainnet` instead of `true`, and that begets the question of where can it find `.mainnet`.  But we don't need to answer that question to solve the problem at hand.

Reply:
Also, `IndexMap` isn't in the rust standard library.

Reply:
Gotcha, makes sense!